### ISSUE TITLE
variable name gets an added prefix when code has qualified Prelude import

### ISSUE THREAD

**Issue by DougBurke** [ndmitchell/hlint#231]
With the program

```
% cat Delme.hs
module Main (main) where

import qualified Prelude as P
import Prelude ((==))

main :: P.IO ()
main = do
  let ans xs = if P.length xs == 0 then "foo" else "bar"
  P.putStrLn (ans [])
```

the hlint response is

```
% hlint Delme.hs
Delme.hs:8:19: Suggestion: Use null
Found:
  P.length xs == 0
Why not:
  P.null P.xs
Note: increases laziness

1 hint
```

Note that `xs` has changed to `P.xs`, which is incorrect. This is with

```
% hlint --version
HLint v1.9.31, (C) Neil Mitchell 2006-2016
```

**Comment by ndmitchell** [comment]
That's awful - thanks for spotting it. I'll take a look.

**Comment by ndmitchell** [comment]
Fixed. I was performing unqualification on the expression after substitution, meaning things like `xs` were qualified. I now perform unqualification before substitution, which fixes it.

**Comment by ndmitchell** [comment]
I intend to make a fresh release tonight.

**Comment by DougBurke** [comment]
Ta muchly.

**Comment by ndmitchell** [comment]
Done, available as hlint-1.9.32
